data/transactions/logic/assembler.go

```diff
@@ -1062,15 +1064,50 @@ func asmDefault(ops *OpStream, spec *OpSpec, args []string) error {
 	return nil
 }
 
+func typeSwap(ops *OpStream, args []string) (StackTypes, StackTypes) {
+	topTwo := oneAny.plus(oneAny)
+	top := len(ops.typeStack) - 1
+	if top > 0 {
+		topTwo[1] = ops.typeStack[top]
+		if top > 1 {
```

Comment:
if len(ops.TypeStack)==2, this will never be true.

Reply:
Yes, I think that's a bug.  I'll add a test and fix.